### PR TITLE
Added async support to 3D plotting in TensorFlow

### PR DESCRIPTION
**Pull request by jonashen**

### CODE REVIEW

**Comment by ghost** [comment]
Sorry about closing the request, I was cleaning some of my branches and this request closed for some reason. I restored it back. Sorry for the inconvenience.

**Review by CatherineSue — APPROVED** [review]

File: sandbox/rocky/tf/plotter/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from .plotter import *
```
Comment (ryanjulian) [inline sandbox/rocky/tf/plotter/__init__.py:1]: PEP8: no relative imports

File: sandbox/rocky/tf/plotter/plotter.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import atexit
+from queue import Empty, Queue
+from threading import Thread
+from rllab.sampler.utils import rollout
```
Comment (ryanjulian) [inline sandbox/rocky/tf/plotter/plotter.py:4]: PEP8: import grouping

File: sandbox/rocky/tf/plotter/plotter.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                        try:
+                            msg = queue.get_nowait()
+                            msgs[msg[0]] = msg[1:]
+                        except Empty:
+                            break
+                    if 'stop' in msgs:
+                        break
+                    elif 'update' in msgs:
+                        env, policy = msgs['update']
+                        # env.start_viewer()
```
Comment (ryanjulian) [inline sandbox/rocky/tf/plotter/plotter.py:40]: please don't commit commented-out code

File: sandbox/rocky/tf/algos/batch_polopt.py [hunk 4]
```diff
@@ -6,7 +6,8 @@
 from sandbox.rocky.tf.samplers.batch_sampler import BatchSampler
 from sandbox.rocky.tf.samplers.vectorized_sampler import VectorizedSampler
 from rllab.sampler.utils import rollout
```
Comment (ryanjulian) [inline sandbox/rocky/tf/algos/batch_polopt.py:8]: PEP8: import grouping

Note: we put rllab. and sandbox. packages in separate groups.

File: sandbox/rocky/tf/algos/batch_polopt.py [hunk 5]
```diff
@@ -88,6 +89,10 @@ def __init__(
 
     def start_worker(self):
         self.sampler.start_worker()
+        if self.plot:
+            plotter.init_plot(self.env, self.policy, self.max_path_length)
+            # Call rollout once to display the window
+            # rollout(self.env, self.policy, animated=True, max_path_length=self.max_path_length)
```
Comment (ryanjulian) [inline sandbox/rocky/tf/algos/batch_polopt.py:95]: please don't commit commented-out code.

is there any way we can do this in `plotter.init_plot()` to make the API simpler?

**Review by ryanjulian — CHANGES_REQUESTED** [review]

**Pull request closed without merging**